src/client/eppo-client.ts

```diff
@@ -372,147 +309,89 @@ export default class EppoClient implements IEppoClient {
     flagKey: string,
     // eslint-disable-next-line @typescript-eslint/no-explicit-any
     subjectAttributes: Record<string, any> = {},
+    defaultValue: EppoValue,
     assignmentHooks: IAssignmentHooks | undefined,
     obfuscated: boolean,
-    valueType?: ValueType,
+    expectedVariationType: VariationType,
   ): EppoValue {
-    const { allocationKey, assignment, holdoutKey, holdoutVariation } = this.getAssignmentInternal(
-      subjectKey,
-      flagKey,
-      subjectAttributes,
-      assignmentHooks,
-      obfuscated,
-      valueType,
-    );
-    assignmentHooks?.onPostAssignment(flagKey, subjectKey, assignment, allocationKey);
-
-    if (!assignment.isNullType() && allocationKey !== null)
-      this.logAssignment(
-        flagKey,
-        allocationKey,
-        assignment,
+    try {
+      const result = this.getAssignmentDetail(
         subjectKey,
-        holdoutKey,
-        holdoutVariation,
+        flagKey,
         subjectAttributes,
+        expectedVariationType,
+        obfuscated,
       );
 
-    return assignment;
+      if (!result.variation) {
+        return defaultValue;
+      }
+
+      return EppoValue.generateEppoValue(result.variation.value, expectedVariationType);
+    } catch (error) {
+      return this.rethrowIfNotGraceful(error, defaultValue);
```

Comment:
Since this is now the only place we use `rethrowIfNotGraceful` consider just in-lining the code and getting rid of that extraneous function. At a minimum, move it's position under this one.

Reply:
Moved function below -- I do like keeping the extra complexity out of the main function for readability